lib/ansible/modules/windows/win_disk_facts.ps1

```diff
@@ -0,0 +1,132 @@
+#!powershell
+
+# Copyright: (c) 2017, Marc Tschapek <marc.tschapek@itelligence.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version 2.0
+
+# Create a new result object
+$result = @{
+    changed = $false
+    ansible_facts = @{
+        ansible_disk = @{
+        } 
+    }
+}
+
+# Search disks
+try {
+    $disk = Get-Disk
+} catch {
+    Fail-Json -obj $result -message "Failed to search the disks on the target: $($_.Exception.Message)"
+}
+if ($disk) {
+    [string]$diskcount = $disk | Measure-Object | Select-Object  -ExpandProperty Count
+    $result.ansible_facts.ansible_disk.total_disks_found = $diskcount
+    $i = 1
+    foreach ($disks in $disk) {
```

Comment:
This should probably be worded in the opposite way, you are looping through `$disks` to get a `$disk` not the other way round.

Reply:
Absolutely right, changed that